codec-http2/src/main/java/io/netty/handler/codec/http2/HpackEncoder.java

```diff
@@ -86,7 +85,7 @@ public HpackEncoder(boolean ignoreMaxHeaderListSize) {
     public HpackEncoder(boolean ignoreMaxHeaderListSize, int arraySizeHint) {
         this.ignoreMaxHeaderListSize = ignoreMaxHeaderListSize;
         maxHeaderTableSize = DEFAULT_HEADER_TABLE_SIZE;
-        maxHeaderListSize = DEFAULT_HEADER_LIST_SIZE;
+        maxHeaderListSize = MAX_HEADER_LIST_SIZE;
```

Comment:
if you look at the comments for `DEFAULT_HEADER_LIST_SIZE` it describes that although the value is arbitrary it is not un-intentional, and technically not in violation of the spec. Having unlimited header size may leave the server exposed to memory exhaustion before it has the chance to communicate a lower limit. I think I would prefer to have the default configurable rather than just making it infinite. What about extracting the value from `AbstractHttp2ConnectionHandlerBuilder#initialSettings`?

Reply:
@Scottmitch Are you saying there is possible scenario in the HTTP/2 protocol in which the receiver may be reading headers from the sender *before* the receiver has sent a settings frame to the sender?

Reply:
I guess I should say: ... *before* the receiver's settings frame has been processed by the sender?

Reply:
@Scottmitch ping

Reply:
doh ... I thought this was in the decoder. I took a look at the decoder and looks like we are already using my suggestion of inferring the value from the initial settings. Sorry for the noise  ... lgtm.